app/src/main/java/org/oppia/android/app/resumelesson/ResumeLessonFragment.kt

```diff
@@ -0,0 +1,88 @@
+package org.oppia.android.app.resumelesson
+
+import android.content.Context
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import org.oppia.android.app.fragment.InjectableFragment
+import javax.inject.Inject
+
+/** Fragment that allows the user to resume a saved exploration. */
+class ResumeLessonFragment :
+  InjectableFragment() {
+
+  companion object {
+    private const val RESUME_LESSON_FRAGMENT_INTERNAL_PROFILE_ID_KEY =
+      "ResumeExplorationFragmentPresenter.resume_exploration_fragment_internal_profile_id"
+    private const val RESUME_LESSON_FRAGMENT_TOPIC_ID_KEY =
+      "ResumeExplorationFragmentPresenter.resume_exploration_fragment_topic_id"
+    private const val RESUME_LESSON_FRAGMENT_STORY_ID_KEY =
+      "ResumeExplorationFragmentPresenter.resume_exploration_fragment_story_id"
+    private const val RESUME_LESSON_FRAGMENT_EXPLORATION_ID_KEY =
+      "ResumeExplorationFragmentPresenter.resume_Lesson_fragment_exploration_id"
+    private const val RESUME_LESSON_FRAGMENT_BACKFLOW_SCREEN_KEY =
+      "ResumeLessonFragmentPresenter.resume_lesson_fragment_backflow_screen"
+
+    /** Creates new instance of [ResumeLessonFragment].
+     * @param internalProfileId is used by the ResumeLessonFragment to retrieve saved checkpoint
+     * @param explorationId is used by the ResumeLessonFragment to retrieve saved checkpoint
+     */
+    fun newInstance(
+      internalProfileId: Int,
+      topicId: String,
+      storyId: String,
+      explorationId: String,
+      backflowScreen: Int
+    ): ResumeLessonFragment {
+      val resumeLessonFragment = ResumeLessonFragment()
+      val args = Bundle()
+      args.putInt(RESUME_LESSON_FRAGMENT_INTERNAL_PROFILE_ID_KEY, internalProfileId)
+      args.putString(RESUME_LESSON_FRAGMENT_TOPIC_ID_KEY, topicId)
+      args.putString(RESUME_LESSON_FRAGMENT_STORY_ID_KEY, storyId)
+      args.putString(RESUME_LESSON_FRAGMENT_EXPLORATION_ID_KEY, explorationId)
+      args.putInt(RESUME_LESSON_FRAGMENT_BACKFLOW_SCREEN_KEY, backflowScreen)
+      resumeLessonFragment.arguments = args
+      return resumeLessonFragment
+    }
+  }
+
+  @Inject
+  lateinit var resumeLessonFragmentPresenter: ResumeLessonFragmentPresenter
+
+  override fun onAttach(context: Context) {
+    super.onAttach(context)
+    fragmentComponent.inject(this)
+  }
+
+  override fun onCreateView(
+    inflater: LayoutInflater,
+    container: ViewGroup?,
+    savedInstanceState: Bundle?
+  ): View? {
+    val internalProfileId =
+      arguments?.getInt(RESUME_LESSON_FRAGMENT_INTERNAL_PROFILE_ID_KEY, -1)!!
```

Comment:
Prefer using checkNotNull with an error rather than ``!!``. Ditto elsewhere in the PR.